### PR TITLE
wgsl: Add AbstractFloat `round` execution tests

### LINKED ISSUE #2509: wgsl: AbstractFloat execution test for `round` builtin
Add an AbstractFloat execution test for the `round` builtin.
```
@const @must_use fn round(e: T) -> T
S is AbstractFloat
T is S or vecN<S>
```
Result is the integer k nearest to e, as a floating point value.
When e lies halfway between integers k and k + 1, the result is k when k is even, and k + 1 when k is odd.
[Component-wise](https://www.w3.org/TR/WGSL/#component-wise) when T is a vector.

Accuracy is `Correctly rounded`

See https://www.w3.org/TR/WGSL/#round-builtin

### PR DESCRIPTION
**Pull request by zoddicus**
Fixes #2509

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/shader/execution/expression/call/builtin/round.spec.ts [hunk 1]
```diff
@@ -43,15 +48,30 @@ export const d = makeCaseCache('round', {
       FP.f16.roundInterval
     );
   },
+  abstract: () => {
+    return FP.abstract.generateScalarToIntervalCases(
+      [
+        reinterpretU64AsF64(0x8000_0000_0000_0000n), // https://github.com/gpuweb/cts/issues/2766
```
Comment (dneto0) [inline src/webgpu/shader/execution/expression/call/builtin/round.spec.ts:54]: This is a reinterpret cast, so this generates  -0.0 in f64.

I don't think that's what you mean.

**Review by dneto0 — CHANGES_REQUESTED** [review]
I think the "large int value" case is not exercised the way it was intended.

File: src/webgpu/shader/execution/expression/call/builtin/round.spec.ts (hunk 1 above)
Comment (dneto0) [reply src/webgpu/shader/execution/expression/call/builtin/round.spec.ts:54]: The point of 2766 was to test integers that are larger in magnitude than can fit inside the mantissa bits.  So for f64 that would be something in excess of 2** 53 ish.

Comment (zoddicus) [reply src/webgpu/shader/execution/expression/call/builtin/round.spec.ts:54]: Correct, thanks for catching this.

Comment (zoddicus) [reply src/webgpu/shader/execution/expression/call/builtin/round.spec.ts:54]: arg... I updated the unittests, but not the CTS test itself :face_exhaling:

Comment (zoddicus) [reply src/webgpu/shader/execution/expression/call/builtin/round.spec.ts]: Updated

**Comment by zoddicus** [comment]
@dneto0 PTAL, I think this should be fixed now

**Review by jzm-intel — APPROVED** [review]
Looks good to me, thanks

**Pull request merged**